### PR TITLE
Use tests from setup.py

### PR DESCRIPTION
**Pull request by bwrsandman**
An example of running tests directly from setup.py, no run_tests.py required.

### CODE REVIEW

**Comment by bwrsandman** [comment]
Tests failing due to pep8 errors in this code. ;)
I seems the same tests from `run_tests.py` are run by `python2 setup.py test`

**Comment by nbessi** [comment]
Thanks,

Yes :laughing:  we may remove run_test.py but before merging I will merge this with MP with new API branch in order to see if it works

**Comment by bwrsandman** [comment]
In many places, I put `# NOQA`.
That is not the way to go about this.
But without knowing the reason behind some of the unused imports, I have not looked further into them.
Just note that # NOQA is like # FIXME for pep8.

**Comment by bwrsandman** [comment]
OK, fixed the pep8 style.

File: setup.py [hunk 1]
```diff
@@ -1,14 +1,16 @@
 # -*- coding: utf-8 -*-
-#Taken from pep8-naming flake8 extention
+# Taken from pep8-naming flake8 extention
```
Comment (florentx) [inline setup.py]: typo: extension

**Comment by bwrsandman** [comment]
Thanks @florentx, I also did a spell-check

**Pull request closed without merging**